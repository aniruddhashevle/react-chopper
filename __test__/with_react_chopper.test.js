import React from "react";
import renderer from "react-test-renderer";
import rcp from "../src/index";
import { shallow, mount, render } from "enzyme";

class WithReactChopper extends React.Component {
  constructor(props) {
    super(props);
    this.state = { firstname: "chopper" };
    this.model = rcp(this.state, this);
  }

  render() {
    return <div>Hello react {this.model.firstname}</div>;
  }

  componentDidMount() {
    this.model.firstname = "chopper changed";
  }
}

it("renders component using react-chopper instead of setstate", () => {
  const tree = renderer.create(<WithReactChopper />).toJSON();
  expect(tree).toMatchSnapshot();
});