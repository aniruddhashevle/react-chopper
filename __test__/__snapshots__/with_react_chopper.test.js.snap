// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component using react-chopper instead of setstate 1`] = `
<div>
  Hello react 
  chopper changed
</div>
`;
